src/test/java/BaseTest.java

```diff
@@ -1,11 +1,126 @@
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+    }
+    @AfterMethod
+    public void closeBrowser() {
+        driver.quit();
+    }
+
+    public void openLoginUrl() {
+        driver.get(url);
+    }
+
+    public void enterEmail(String email) {
+        WebElement emailInput = driver.findElement(By.cssSelector("input[type='email']"));
+        emailInput.click();
+        emailInput.clear();
+        emailInput.sendKeys(email);
+    }
+
+    public void enterPassword(String password) {
+        WebElement passwordInput = driver.findElement(By.cssSelector("input[type='password']"));
+        passwordInput.click();
+        passwordInput.clear();
+        passwordInput.sendKeys(password);
+    }
+
+    public void clickSubmit() {
+        WebElement submitBtn = driver.findElement(By.cssSelector("button[type='submit']"));
+        submitBtn.click();
+    }
+
+    //Login method
+    public void loginWithCorrectCred() {
+        openLoginUrl();
+        enterEmail("emiliano.castillo@testpro.io");
+        enterPassword("te$t$tudent");
+        clickSubmit();
+    }
+
+
+
+    //Profile Test Helper Functions
+    public void clickAvatarIcon() {
+        WebElement avatarIcon = driver.findElement(By.cssSelector("img.avatar"));
+        avatarIcon.click();
+    }
+    public void provideCurrentPassword(String password) {
+        WebElement currentPassword = driver.findElement(By.cssSelector("[name='current_password']"));
+        currentPassword.clear();
+        currentPassword.sendKeys(password);
+    }
+    public void clickSaveButton() {
+        WebElement saveButton = driver.findElement(By.cssSelector("button.btn-submit"));
+        saveButton.click();
+    }
+    public void provideProfileName(String randomName) {
+        WebElement profileName = driver.findElement(By.cssSelector("[name='name']"));
+        profileName.clear();
+        profileName.sendKeys(randomName);
+    }
+    public String generateRandomName () {
+        return UUID.randomUUID().toString().replace("-", "");
+    }
+
+    //Homework 17
+
+    //This method will click on search field and search whatever inputText we choose
+    public void searchForSong (By inputLocator, String inputText) {
+        WebElement searchField = driver.findElement(inputLocator);
+        searchField.click();
+        searchField.clear();
+        searchField.sendKeys(inputText);
+    }
+    public void enterSongIntoSearchField() {
+        searchForSong(By.cssSelector("input[type='search']"), "Dark Days");
+    }
+    //Click view all btn
+    public void clickViewAll () {
+        WebElement viewAllBtn = driver.findElement(By.cssSelector("[data-test='view-all-songs-btn']"));
+        viewAllBtn.click();
+    }
+    public void clickFirstSong () {
+        WebElement firstSong = driver.findElement(By.xpath("//*[@id='songResultsWrapper']/div/div/div[1]/table/tr[1]/td[2]"));
```

Comment:
In this `firstSong` selector too long and it is Absolute xpath.
When you define selector best practices for locators should be
- Start with the following order: id, unique attribute, css, xpath
- Never use Absolute paths, always use relative ones
- Your locator should be unique 
- Keep your locator simple and robust
- Don’t locate elements by visual labels (not reliable, may change often)

//HINT: suggestion `WebElement firstSong = driver.findElement(By.xpath("(//*[@id='songResultsWrapper']//tr[@class='song-item'])[1]"));`
or `WebElement firstSong = driver.findElement(By.cssSelector("#songResultsWrapper [class='song-item']:nth-child(1)"));`